File: backend/app/services/email_service.py
import logging
import os

logger = logging.getLogger(__name__)

FRONTEND_URL = os.getenv("FRONTEND_URL", "http://localhost:5173")


def send_password_reset_email(email: str, reset_token: str) -> None:
    reset_url = f"{FRONTEND_URL}/reset-password?token={reset_token}"

    smtp_host = os.getenv("SMTP_HOST")
    if smtp_host:
        # SMTP integration point — wire when credentials are configured.
        logger.info("Password reset email would be sent to %s (SMTP not fully wired).", email)
        logger.info("Reset URL: %s", reset_url)
        return

    logger.info(
        "[HIRELENS] Password reset for %s — configure SMTP to send email. Reset URL: %s",
        email,
        reset_url,
    )

File: backend-node/src/services/emailService.js
import nodemailer from 'nodemailer'
import { env } from '../config/env.js'

export async function sendPasswordResetEmail(email, resetToken) {
  const resetUrl = `${env.frontendUrl}/reset-password?token=${resetToken}`

  if (!env.smtpHost) {
    console.info(
      `[HIRELENS] Password reset for ${email} — configure SMTP to send email. Reset URL: ${resetUrl}`,
    )
    return
  }

  const transporter = nodemailer.createTransport({
    host: env.smtpHost,
    port: env.smtpPort,
    secure: env.smtpPort === 465,
    auth: env.smtpUser
      ? { user: env.smtpUser, pass: env.smtpPass }
      : undefined,
  })

  await transporter.sendMail({
    from: env.smtpFrom || env.smtpUser || 'noreply@hirelens.local',
    to: email,
    subject: 'HIRELENS password reset',
    text: `Reset your password: ${resetUrl}`,
    html: `<p>Reset your password:</p><p><a href="${resetUrl}">${resetUrl}</a></p>`,
  })
}
